File: client/flutter/lib/service/moment.dart
import 'package:app/generated/protobuf/content/moment.service.pbgrpc.dart';
import 'package:app/model/moment/moment.service.dart';


import 'package:app/utils/dio.dart';
import 'package:get/get.dart';
import 'package:grpc/grpc.dart';
import 'package:app/model/response.dart';

class MomentClient extends GetxService {
  late final MomentServiceClient stub;

  MomentClient() : super() {
    final channel = ClientChannel(
      'hoper.xyz',
      port: 8090,
      options: const ChannelOptions(credentials: ChannelCredentials.insecure()),
    );
    this.stub = MomentServiceClient(channel);
  }

  Future<MomentListResponse$?> getMomentList(int pageNo, pageSize) async {
    print(pageNo);
    var api = '/v1/moment?pageNo=$pageNo&pageSize=$pageSize';

    try {
      var response = await httpClient.get(api);
      return MomentListResponse$.fromJson(response.getData());
    } catch (exception) {
      return null;
    }
  }
}

File: client/web/src/h5/service/moment.ts
import { MomentServiceClient } from "@generated/protobuf-ts/content/moment.service.client";
import type { MomentListRep } from "@generated/protobuf-ts/content/moment.service";
import { GrpcWebFetchTransport } from "@protobuf-ts/grpcweb-transport";
import { showFailToast } from "vant";
import type { RpcError } from "@protobuf-ts/runtime-rpc";

const momentClient = new MomentServiceClient(
  new GrpcWebFetchTransport({
    baseUrl: "https://grpc.hoper.xyz",
  })
);

export async function momentList(
  pageNo: number,
  pageSize: number
): Promise<MomentListRep> {
  try {
    const { response, status } = await momentClient.list({ pageNo, pageSize });
    console.log(status);
    return response;
  } catch (e) {
    const rpcError = e as RpcError;
    showFailToast(decodeURI(rpcError.message));
    return Promise.reject(rpcError);
  }
}
